### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by davetaflin**
## Proposed Changes
Added parallel output of Tecplot binary .szplt files.
Updated TecIO source to the latest version.

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [X] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.

### CODE REVIEW

**Comment by economon** [comment]
@davetaflin : great contribution! Thanks to you and @ScottImlay at Tecplot for the effort. Nice to have you as part of the community.

File: Common/include/mpi_structure.inl [hunk 1]
```diff
@@ -597,6 +609,12 @@ inline void CBaseMPIWrapper::Alltoall(void *sendbuf, int sendcount, Datatype sen
   CopyData(sendbuf, recvbuf, recvcount, sendtype);
 }
 
+inline void CBaseMPIWrapper::Alltoallv(void *sendbuf, int *sendcounts, int *sdispls, Datatype sendtype,
```
Comment (economon) [inline Common/include/mpi_structure.inl]: @talbring : does everything here look ok for including Alltoallv() in the wrapper?

File: externals/tecio/examples/brickpartitioned/brickpartitioned.vcxproj [hunk 2]
```diff
@@ -136,7 +136,7 @@
     </Midl>
     <ClCompile>
       <Optimization>Disabled</Optimization>
-      <AdditionalIncludeDirectories>..\..\..\..\include;C:\Program Files\Microsoft MPI\Inc;C:\Program Files (x86)\Microsoft SDKs\MPI\Include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>..\..\..\..\include;..\..\teciompisrc;C:\Program Files\Microsoft MPI\Inc;C:\Program Files (x86)\Microsoft SDKs\MPI\Include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (economon) [inline externals/tecio/examples/brickpartitioned/brickpartitioned.vcxproj]: Can the .vcxproj files be removed from version control?

Comment (davetaflin) [reply externals/tecio/examples/brickpartitioned/brickpartitioned.vcxproj]: Yep, I'll remove those and related Visual Studio-only files.

**Comment by davetaflin** [comment]
@economon to your question above:
> Thanks again, @davetaflin !
> 
> Just a couple of comments/questions first in the review. Also, when users use SU2_SOL to generate solution files (instead of during runtime with your new implementation), will the older serial implementation still behave as expected for writing Tecplot files?

SU2_SOL still calls COutput::SetBaselineResult_Files, which, for TECPLOT_BINARY output, calls SetTecplotBinary_DomainMesh and SetTecplotBinary_DomainSolution. These are unchanged here, and in my testing, continue to output mesh and solution .szplt files (changed by Scott Imlay to .szplt from .plt).

**Comment by economon** [comment]
Sounds like this one is more or less ready too.. 

@talbring, can you please check the changes to the MPI wrapper in this PR when you have some time?

**Pull request merged**